model/modelprocessor/datastream.go

```diff
@@ -87,6 +88,19 @@ func isRUMAgentName(agentName string) bool {
 	return false
 }
 
+func getAppLogsDataset(event *model.APMEvent) string {
+	serviceName := event.Service.Name
+	if serviceName == "" {
+		serviceName = logDefaultServiceName
+	}
```

Comment:
Even though ServiceName will be always supplied by the agents, added a safeguard here as I observed the `model.APMEvent` can omit the ServiceName and it can create datastreams like `logs-apm.app.-*`. But if we are confident that this will never happen, we can skip this.